colossus-metrics/src/main/scala/colossus/metrics/Aggregation.scala

```diff
@@ -38,20 +38,23 @@ object TagSelector {
 }
 
 sealed trait AggregationType {
-  def aggregate(values: Seq[Long]): Long
+  def aggregate(values: Seq[MetricValue]): RawMetricValue
 }
 object AggregationType {
   case object Sum extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum
+    def aggregate(values: Seq[MetricValue]): RawMetricValue = values.foldLeft(0L){case (build: RawMetricValue, next: MetricValue) => build + next.value}
   }
   case object Average extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum / values.size
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (build, next) => build + next.value} / values.size
```

Comment:
Yeah there should be a check, though apparently it's never actually happened because it's been like that since forever and I've never seen an exception thrown.